vault/resource_database_secret_backend_static_role_test.go

```diff
@@ -0,0 +1,160 @@
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "rotation_period", "3600"),
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "rotation_statements.0", "SELECT 1;"),
+				),
+			},
+			{
+				Config: testAccDatabaseSecretBackendStaticRoleConfig_updated(name, username, dbName, backend, connURL),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "name", name),
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "backend", backend),
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "username", username),
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "db_name", dbName),
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "rotation_period", "1800"),
+					resource.TestCheckResourceAttr("vault_database_secret_backend_static_role.test", "rotation_statements.0", "SELECT 2;"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatabaseSecretBackendStaticRoleCheckDestroy(s *terraform.State) error {
+	client := testProvider.Meta().(*api.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vault_database_secret_backend_static_role" {
+			continue
+		}
+		secret, err := client.Logical().Read(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if secret != nil {
+			return fmt.Errorf("sttatic role %q still exists", rs.Primary.ID)
+		}
+	}
+	return nil
+}
+
+func testAccDatabaseSecretBackendStaticRoleConfig_basic(name, username, db, path, connURL string) string {
+	return fmt.Sprintf(`
+resource "vault_mount" "db" {
+  path = "%s"
+  type = "database"
+}
+
+resource "vault_database_secret_backend_connection" "test" {
+  backend = "${vault_mount.db.path}"
+  name = "%s"
+  allowed_roles = ["*"]
+
+  postgresql {
+	  connection_url = "%s"
+  }
+}
+
+resource "vault_database_secret_backend_static_role" "test" {
+  backend = "${vault_mount.db.path}"
+  db_name = "${vault_database_secret_backend_connection.test.name}"
+  name = "%s"
+  username = "%s"
+  rotation_period = 3600
+  rotation_statements = ["SELECT 1;"]
+}
+`, path, db, connURL, name, username)
+}
+
+func testAccDatabaseSecretBackendStaticRoleConfig_updated(name, username, db, path, connURL string) string {
+	return fmt.Sprintf(`
+resource "vault_mount" "db" {
+  path = "%s"
+  type = "database"
+}
+
+resource "vault_database_secret_backend_connection" "test" {
+  backend = "${vault_mount.db.path}"
+  name = "%s"
+  allowed_roles = ["*"]
+
+  postgresql {
+	  connection_url = "%s"
+  }
+}
+
+resource "vault_database_secret_backend_static_role" "test" {
+  backend = "${vault_mount.db.path}"
+  db_name = "${vault_database_secret_backend_connection.test.name}"
+  name = "%s"
+  username = "%s"
+  rotation_period = 1800
+  rotation_statements = ["SELECT 2;"]
```

Comment:
If we end up marking rotation statements optional, we should probably also add one test where they're not included.